graph/preprocessor.go

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* Alias preprocessor:
+* The set of elements here are meant to process the alias definition portion of task.yaml
+* files. This is done by unmarshaling these elements which will then be added in a hierarchical
+* manner. Note the input must still be valid YAML.
+*
+* TODO:
+* Acquire list of globally accessible image endpoints
+ */
+
+package graph
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+var (
+	errImproperDirectiveLength = errors.New("$ directive can only be overwritten by a single character")
+	errImproperKeyName         = errors.New("alias key names only support alphanumeric characters")
+	errImproperDirectiveChoice = errors.New("overwritten directives may not be alphanumeric characters")
+	directive                  = '$'
+	aliasFormat                = regexp.MustCompile("\\A[a-zA-Z0-9]+\\z")
```

Comment:
No, I mean this is generally faster. I meant for when I am validating the aliases themselves (making sure they are alphanumeric by using the regex instead of looping over each character and checking this?)